mlflow/utils/search_utils.py

```diff
@@ -272,7 +297,20 @@ def _does_run_match_clause(cls, run, sed):
                                   error_code=INVALID_PARAMETER_VALUE)
         if lhs is None:
             return False
-        if comparator in cls.filter_ops.keys():
+
+        if comparator in ['like', 'ilike']:
+            # Change value from sql syntax to regex syntax
+            if comparator == 'ilike':
+                value = value.lower()
+                lhs = lhs.lower()
+            if not value.startswith('%'):
+                value = '^' + value
+            if not value.endswith('%'):
+                value = value + '$'
```

Comment:
Is this required for SQLAlchemy? As if query is specified as `tags.algorithm LIKE '%st'` SQL code would only return values ending in **st**. Is this `$` enforcement required?

Reply:
In fact this part of the code is only used by file_store, so the goal here is to convert the sql syntax to a regex syntax so '^' and '$' are used to say that we want to match the beginning or the end of the string.

Reply:
I wonder if store specific syntax parsing and conversion belongs in those specific store. This is the generic API level code parsing and the objective is to be SQL compliant.

Can you move the `^` and `$` handling to `FileStore`?

Reply:
I agree, this should be in the file store the same way the functions to filter sql are in the sql alchemy store. (since this pr: https://github.com/mlflow/mlflow/pull/2030)

But if we follow this logic, `search_utils.filter` and `search_utils._does_run_match_clause` should be entirely in file store since it is only used by this store and the logic is specific for searching in a file. Should I move these two functions to the file store ?

Reply:
The functions you mention used to be used by both file store and sql store, before we pushed down the SQL implementation to the database. Now you are right, those should exist only in the file store. I think it is reasonable to move them in this PR.